app/controllers/escorts_controller.rb

```diff
@@ -1,6 +1,7 @@
 class EscortsController < ApplicationController
   helper_method :escort
   before_action :redirect_if_missing_data, only: :show
+  before_action :redirect_if_issued_or_cancelled_escort, only: [:confirm_cancel, :cancel]
```

Comment:
If tomorrow there's a third state that also matches this criteria, this method name will need to be changed. The name of the method shouldn't depend on the internal logic of it.

Consider instead something like `:redirect_if_not_cancellable`.